keras_rs/src/layers/embedding/base_distributed_embedding.py

```diff
@@ -0,0 +1,566 @@
+import collections
+from typing import Any, Optional, Sequence, Union
+
+import keras
+
+from keras_rs.src import types
+from keras_rs.src.layers.embedding import distributed_embedding_config
+from keras_rs.src.layers.embedding import embed_reduce
+from keras_rs.src.utils import keras_utils
+
+FeatureConfig = distributed_embedding_config.FeatureConfig
+TableConfig = distributed_embedding_config.TableConfig
+EmbedReduce = embed_reduce.EmbedReduce
+
+
+# Placeholder for tf.tpu.experimental.embedding.FeatureConfig.
+TfFeatureConfig = Any
+
+# Placeholder for tf.tpu.experimental.embedding._Optimizer (not exported).
+TfTpuOptimizer = Any
+
+# Placeholder for tf.tpu.experimental.embedding.SparseCoreEmbeddingConfig.
+TfSparseCoreEmbeddingConfig = Any
+
+
+SUPPORTED_PLACEMENTS = ("auto", "default_device", "sparsecore")
+
+
+PlacementAndPath = collections.namedtuple(
+    "PlacementAndPath", ["placement", "path"]
+)
+
+
+class DistributedEmbedding(keras.layers.Layer):
+    """DistributedEmbedding, a layer for accelerated large embedding lookups.
+
+    Args:
+        feature_configs: A nested structure of `FeatureConfig` or
+            `tf.tpu.experimental.embedding.FeatureConfig`. All the configuration
+            objects must be of the same type.
+        table_stacking: The table stacking to use. `None` means no table
+            stacking. `"auto"` means to stack tables automatically. A list of
+            table names or list of lists of table names means to stack the
+            tables in the inner lists together. Note that table stacking is not
+            supported on older TPUs, in which case the default value of `"auto"`
+            will be interpreted as no table stacking. Also note that this
+            argument cannot be used when passing
+            `tf.tpu.experimental.embedding.FeatureConfig` instances for the
+            `feature_configs` argument. Instead, the `disable_table_stacking`
+            attribute of the`sparse_core_embedding_config` argument should be
+            used.
+        optimizer: String, `keras.optimizers.Optimizer`, or one of the
```

Comment:
Are these TF-specific arguments necessary in the base class?

We could also just check `kwargs` for backend-specific args, and if it's not empty at this point it means they weren't properly handled.

Reply:
The issue is that there is only one documentation, and it will come from this class. So that's the only way to document these arguments.

Reply:
You can add it to the documentation body, but keep it out of the actual function args.

Reply:
Let me remove them for now. I haven't tested them in a while.